vidarr-server/src/test/java/ca/on/oicr/gsi/vidarr/server/VeryBadDataIntegrationTest.java

```diff
@@ -0,0 +1,224 @@
+    config.setWorkflowEngines(new HashMap<>());
+    config.setOutputProvisioners(new HashMap<>());
+    config.setRuntimeProvisioners(new HashMap<>());
+    config.setTargets(new HashMap<>());
+    config.setUnloadDirectory(unloadDirectory.getRoot().getAbsolutePath());
+    return config;
+  }
+
+  @BeforeClass
+  public static void setup() throws SQLException {
+    TimeZone.setDefault(TimeZone.getTimeZone("America/Toronto"));
+    config = getTestServerConfig(pg);
+    main = new Main(config);
+    main.startServer(main);
+    RestAssured.baseURI = config.getUrl();
+    RestAssured.port = config.getPort();
+    RestAssured.enableLoggingOfRequestAndResponseIfValidationFails();
+    defaultParser = Parser.TEXT;
+    MAPPER.registerModule(new JavaTimeModule());
+  }
+
+  @Before
+  public void cleanAndMigrateDB() {
+    final var simpleConnection = new PGSimpleDataSource();
+    simpleConnection.setServerNames(new String[] {config.getDbHost()});
+    simpleConnection.setPortNumbers(new int[] {config.getDbPort()});
+    simpleConnection.setDatabaseName(config.getDbName());
+    simpleConnection.setUser(config.getDbUser());
+    simpleConnection.setPassword(config.getDbPass());
+    var fw = Flyway.configure().dataSource(simpleConnection);
+    fw.load().clean();
+    fw.locations("classpath:db/migration").load().migrate();
+    // we do this because Flyway on its own isn't finding the test data, and it dies when you
+    // try to give it classpath + filesystem locations in one string. We ignore the "missing"
+    // migrations (run in the migrate() call above).
+    fw.locations("filesystem:src/test/resources/db/migration/")
+        .ignoreMissingMigrations(true)
+        .load()
+        .migrate();
+  }
+
+  public List<String> getNaughtyStringList() {
+    List<String> blns = NaughtyStrings.getStrings();
+    // These indices correspond with strings here
+    // https://github.com/minimaxir/big-list-of-naughty-strings/blob/master/blns.json
+    List<Integer> indices =
+        IntStream.concat(IntStream.range(0, 31), IntStream.range(194, 202))
+            .boxed()
+            .collect(Collectors.toList());
+    // Filtering original list to only get strings we care about
+    return IntStream.range(0, blns.size())
+        .filter(indices::contains)
+        .mapToObj(blns::get)
+        .collect(Collectors.toList());
+  }
+
+  @Test
+  public void getWorkflow() {
+    List<String> filteredBlns = getNaughtyStringList();
+    filteredBlns.forEach(
+        (naughtyString) -> {
+          get("/api/workflow/{name}", naughtyString).then().assertThat().statusCode(404);
+        });
+  }
+
+  @Test
+  public void addWorkflow() throws JsonProcessingException {
+    List<String> filteredBlns = getNaughtyStringList();
+    var noParamWorkflow = MAPPER.writeValueAsString(new HashMap<>());
+
+    // Vidarr doesn't actually have any restrictions on the name of a workflow so all of these will
+    // succeed
```

Comment:
Should Vidarr have some restrictions on the name of a workflow? Being able to name a workflow 'null' is probably going to hide some weird failure modes isn't it?